beacon-chain/forkchoice/protoarray/optimistic_sync.go

```diff
@@ -0,0 +1,85 @@
+package protoarray
+
+import (
+	"fmt"
+
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/config/params"
+	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+)
+
+// This returns the minimum and maximum slot of the synced_tips tree
+func (f *ForkChoice) boundarySyncedTips() (types.Slot, types.Slot) {
+	f.syncedTips.RLock()
+	defer f.syncedTips.RUnlock()
+
+	min := params.BeaconConfig().FarFutureSlot
+	max := types.Slot(0)
+	for _, slot := range f.syncedTips.validatedTips {
+		if slot > max {
+			max = slot
+		}
+		if slot < min {
+			min = slot
+		}
+	}
+	return min, max
+}
+
+// Optimistic returns true if this node is optimistically synced
```

Comment:
What does it mean to be optimistically synced?

Reply:
It means that we have synced the block as usual, but we haven't validated the execution Payload

Reply:
Please add that to the godoc description, otherwise this commentary doesn't add anymore information to the code.